### PR TITLE
[2.0] SSH2: Remove all static variables as loadable classes may change at runtime.

### PR DESCRIPTION
**Pull request by bantu**

### CODE REVIEW

**Comment by terrafrost** [comment]
How would you unload a class at run time?

**Comment by bantu** [comment]
@terrafrost It's more about adding classes by adding an additional autoloader. Also, the next step would be to make the supported algorithms changeable via method calls.

**Comment by terrafrost** [comment]
Works for me. Go for it.

File: phpseclib/Net/SSH2.php [hunk 1]
```diff
@@ -1167,90 +1167,87 @@ function _key_exchange($kexinit_payload_server)
...
-                );
-            }
-            if (class_exists('\phpseclib\Crypt\TripleDES') === false) {
-                $encryption_algorithms = array_diff(
-                    $encryption_algorithms,
-                    array('3des-ctr', '3des-cbc')
-                );
-            }
-            $encryption_algorithms = array_values($encryption_algorithms);
+        if (class_exists('\phpseclib\Crypt\RC4') === false) {
```
Comment (GrahamCampbell) [inline phpseclib/Net/SSH2.php]: no leading slash should be used - infact it's wrong to include it here

File: phpseclib/Net/SSH2.php [hunk 2]
```diff
@@ -1167,90 +1167,87 @@ function _key_exchange($kexinit_payload_server)
...
+            $encryption_algorithms = array_diff(
+                $encryption_algorithms,
+                array('arcfour256', 'arcfour128', 'arcfour')
+            );
+        }
+        if (class_exists('\phpseclib\Crypt\Rijndael') === false) {
```
Comment (GrahamCampbell) [inline phpseclib/Net/SSH2.php]: same

File: phpseclib/Net/SSH2.php [hunk 3]
```diff
@@ -1167,90 +1167,87 @@ function _key_exchange($kexinit_payload_server)
...
+            $encryption_algorithms = array_diff(
+                $encryption_algorithms,
+                array('aes128-ctr', 'aes192-ctr', 'aes256-ctr', 'aes128-cbc', 'aes192-cbc', 'aes256-cbc')
+            );
+        }
+        if (class_exists('\phpseclib\Crypt\Twofish') === false) {
```
Comment (GrahamCampbell) [inline phpseclib/Net/SSH2.php]: in all places

File: phpseclib/Net/SSH2.php (hunk 1 above)
Comment (bantu) [inline phpseclib/Net/SSH2.php]: @GrahamCampbell I am going to merge this now. I would appreciate if you could file a new PR with information explaining why it's wrong.

**Pull request merged**